src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,167 @@
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const endpointPlanner: string = `${this.resource}/v1.0/planner/plans`;
+    if (args.options.ownerGroupId) {
+      this.getGroupsWithFilter(`?$filter=ID eq '${args.options.ownerGroupId}'`)
+        .then((groups: any[]): void => {
+          if (groups && groups.length > 0) {
+            if (groups.length > 1) {
+              logger.log(`More than one groups found with id ${args.options.ownerGroupId}`);
+              cb();
+            }
+            else {
+              const requestOptions: any = {
+                url: endpointPlanner,
+                headers: {
+                  'accept': 'application/json;odata.metadata=none'
+                },
+                responseType: 'json',
+                data: {
+                  "owner": groups[0].id,
+                  "title": args.options.title
+                }
+              };
+              request
+                .post(requestOptions)
+                .then((res: any): void => {
+                  logger.log(res);
+                  cb();
+                }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+            }
+          }
+          else {
+            logger.log(`Owner group not found with id ${args.options.ownerGroupId}`);
+            cb();
+          }
+        }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+    }
+    else {
+      const ownerGroupNameFilter: string = args.options.ownerGroupName ? `?$filter=DisplayName eq '${encodeURIComponent(args.options.ownerGroupName).replace(/'/g, `''`)}'` : '';
+      this.getGroupsWithFilter(ownerGroupNameFilter)
+        .then((groups: any[]): void => {
+          if (groups && groups.length > 0) {
+            if (groups.length > 1) {
+              logger.log(`More than one groups found with name ${args.options.ownerGroupName}`);
+              cb();
+            }
+            else {
+              const requestOptions: any = {
+                url: endpointPlanner,
+                headers: {
+                  'accept': 'application/json;odata.metadata=none'
+                },
+                responseType: 'json',
+                data: {
+                  "owner": groups[0].id,
+                  "title": args.options.title
+                }
+              };
+              request
+                .post(requestOptions)
+                .then((res: any): void => {
+                  logger.log(res);
+                  cb();
+                }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+            }
+          }
+          else {
+            logger.log(`Owner group not found with name ${args.options.ownerGroupName}`);
+            cb();
+          }
+        }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+    }
+  }
+
+  private async getGroupsWithFilter(ownerGroupFilter: string): Promise<any[]> {
```

Comment:
Rather than returning all groups matching the filter back to the main code, why not change it so that this method returns the group ID or error if no group was found? See https://github.com/pnp/cli-microsoft365/blob/713002b84c3088c7b69a563a51a08ec05295252f/src/m365/teams/commands/channel/channel-get.ts#L71 for an example